### PR TITLE
chore(deps): update

### PR DESCRIPTION
**Pull request by fanatid**
- `futures` #5952 
- `pin-project` #5969 
- `rand` #5953 
- `cargo update --aggressive`

### CODE REVIEW

File: lib/remap-lang/src/value.rs [hunk 1]
```diff
@@ -15,6 +15,7 @@ use std::str::FromStr;
 
 pub use kind::Kind;
 
+#[allow(clippy::large_enum_variant)]
```
Comment (jszwedko) [inline lib/remap-lang/src/value.rs:18]: It'd be nice to add comments for these allows about why we allow them.

**Review by jszwedko — APPROVED** [review]
In general this looks good!

I'd like to see us do a little investigation to decide if the large enum variants are OK or if it might be better to box the large ones. Assuming we are ok with it, it'd be nice to leave comments next to the `#[allow]` blocks about why we are ok with them.

**Comment by jszwedko** [comment]
Noting that I'm fine opening a follow-up issue for investigating the large enum variants since this PR doesn't change that.

File: lib/remap-lang/src/value.rs (hunk 1 above)
Comment (fanatid) [reply lib/remap-lang/src/value.rs:18]: I checked `regex` code for understanding why we actually get this error. So `regex` use [thread_local](https://crates.io/crates/thread_local) for some cache. Version `1.1.0` of `thread_local` was released on `Jan 8, 2021` and one of the changes was inline data to structure from the `Box`:
https://github.com/Amanieu/thread_local-rs/blob/v1.0.1/src/lib.rs#L95
vs
https://github.com/Amanieu/thread_local-rs/blob/v1.1.0/src/lib.rs#L107
Changes were made in this PR: https://github.com/Amanieu/thread_local-rs/pull/22
I'm not sure how this change will affect `vector` memory consumption, so will revert `thread_local` update and create a separate issue for this.

**Comment by jszwedko** [comment]
@fanatid :+1: thanks for investigating!

**Pull request merged**